### PR TITLE
Migrate to terser from uglify for apps | Updates dependencies

### PR DESCRIPTION
**Pull request by anselmbradford**
## Changes

- Migrate to terser from uglify for apps scripts.
- Update to latest `@babel/core`, `@babel/preset-env`, `del`, `gulp`, `gulp-clean-css`, `lightbox2`, `webpack`, `lighthouse`, `protractor-retry`, and `snyk` in the main repo app.
- Update to latest `cf-core` in find-a-housing-counselor app.
- Update to latest `handlebars` in owning-a-home app.
- Update 

## Testing

1. Run `yarn`.
2. Run `gulp clean && gulp build`
3. http://localhost:8000/know-before-you-owe/compare/ lightboxes should work like production (click images).
4. http://localhost:8000/find-a-housing-counselor/ should work like production.
5. http://localhost:8000/owning-a-home/explore-rates/ and enter 600,000 for the loan amount. Text `Based on your loan amount, you may not be eligible for a regular (conforming) conventional loan.…` is a handlebars template and should appear.
6. `gulp audit:perf` should execute.
7. `yarn run test` should execute.

### CODE REVIEW

**Comment by anselmbradford** [comment]
Travis is failing because https://github.com/cfpb/cfgov-refresh/pull/4986 needs to be merged

File: cfgov/unprocessed/apps/find-a-housing-counselor/webpack-config.js [hunk 1]
```diff
@@ -4,14 +4,14 @@
 
 const BROWSER_LIST = require( '../../../../config/browser-list-config' );
 const webpack = require( 'webpack' );
-const UglifyWebpackPlugin = require( 'uglifyjs-webpack-plugin' );
+const TerserPlugin = require( 'terser-webpack-plugin' );
 
 /* Set warnings to true to show linter-style warnings.
    Set mangle to false and beautify to true to debug the output code. */
-const COMMON_UGLIFY_CONFIG = new UglifyWebpackPlugin( {
+const COMMON_UGLIFY_CONFIG = new TerserPlugin( {
```
Comment (contolini) [inline cfgov/unprocessed/apps/find-a-housing-counselor/webpack-config.js]: With the uglify plugin gone maybe change this to `COMMON_MINIFICATION_CONFIG`.

Comment (anselmbradford) [reply cfgov/unprocessed/apps/find-a-housing-counselor/webpack-config.js]: Hmmmmmmmmmmmmmmmmmmmmm good point

Comment (anselmbradford) [reply cfgov/unprocessed/apps/find-a-housing-counselor/webpack-config.js]: It is done in https://github.com/cfpb/cfgov-refresh/pull/4987/commits/25d9f83e5f2ddd24318ccfbcb69a666f46982308
Also, please sign off on https://github.com/cfpb/capital-framework/pull/917

**Pull request merged**